### PR TITLE
Replace azuredeploy []byte with template map[string]interface{}

### LINKED ISSUE #547: replace azuredeploy []byte with template map[string]interface{} in all function signatures
In reviewing https://github.com/openshift/openshift-azure/pull/521 @jim-minter suggested we change
all azuredeploy []byte to map[string]interface{} 

example:
	CreateOrUpdate(ctx context.Context, cs *OpenShiftManagedCluster, azuredeploy []byte, isUpdate bool, Deployer DeployFn) error

We have now changed the custom deploy function to the new parameter, but we should change all of them.

### PR DESCRIPTION
**Pull request by asalkeld**
closes #547

### CODE REVIEW

File: cmd/createorupdate/createorupdate.go [hunk 1]
```diff
@@ -128,7 +129,12 @@ func createOrUpdate(ctx context.Context, oc *v20180930preview.OpenShiftManagedCl
 		return future.WaitForCompletionRef(ctx, deployments.Client())
 	}
 
-	err = p.CreateOrUpdate(ctx, cs, azuredeploy, oldCs != nil, deployer)
+	var azuretemplate map[string]interface{}
```
Comment (jim-minter) [inline cmd/createorupdate/createorupdate.go]: can we have GenerateARM return map[string]interface{} ?

Comment (asalkeld) [reply cmd/createorupdate/createorupdate.go]: sure, i can give that a go

File: pkg/plugin/plugin_test.go [hunk 2]
```diff
@@ -65,12 +64,21 @@ func TestMerge(t *testing.T) {
...
 	armTemplate = testPluginRun(p, newCluster, oldCluster, t)
-	if strings.Contains(string(armTemplate), "\"type\": \"Microsoft.Network/networkSecurityGroups\"") {
+	if hasResourceType(armTemplate, "Microsoft.Network/networkSecurityGroups") {
 		t.Fatalf("networkSecurityGroups should not be applied during cluster upgrade")
 	}
 }
 
-func testPluginRun(p api.Plugin, newCluster *api.OpenShiftManagedCluster, oldCluster *api.OpenShiftManagedCluster, t *testing.T) (armTemplate []byte) {
+func hasResourceType(armTemplate map[string]interface{}, resType string) bool {
+	for _, res := range armTemplate["resources"].([]interface{}) {
```
Comment (jim-minter) [inline pkg/plugin/plugin_test.go]: fyi - I implemented a subset of jsonpath which can be useful for this kind of thing - take a look at our use of pkg/jsonpath

**Comment by jim-minter** [comment]
rebased for you

/lgtm

**Pull request merged**